packages/cli/src/lib/implementation/validate-filter-options.utils.ts

```diff
@@ -55,6 +63,68 @@ export function validateFilterOption(
   }
 }
 
+export function validateFilteredCategories(
+  originalCategories: NonNullable<Filterables['categories']>,
+  filteredCategories: NonNullable<Filterables['categories']>,
+  {
+    onlyCategories,
+    skipCategories,
+    verbose,
+  }: Pick<FilterOptions, 'onlyCategories' | 'skipCategories' | 'verbose'>,
+): void {
+  const skippedCategories = originalCategories.filter(
+    original => !filteredCategories.some(({ slug }) => slug === original.slug),
+  );
+  if (verbose) {
+    skippedCategories.forEach(category => {
+      ui().logger.info(
+        `Category ${category.slug} was removed because all its refs were skipped. Affected refs: ${category.refs
+          .map(ref => `${ref.slug} (${ref.type})`)
+          .join(', ')}`,
+      );
+    });
+  }
+  const invalidArgs = [
+    { option: 'onlyCategories', args: onlyCategories ?? [] },
+    { option: 'skipCategories', args: skipCategories ?? [] },
+  ].filter(({ args }) =>
+    args.some(arg => skippedCategories.some(({ slug }) => slug === arg)),
+  );
+  if (invalidArgs.length > 0) {
+    throw new OptionValidationError(
+      invalidArgs
+        .map(
+          ({ option, args }) =>
+            `The --${option} argument references skipped categories: ${args.join(', ')}`,
+        )
+        .join('. '),
+    );
+  }
```

Comment:
I wouldn't throw validation errors in this case. Skipping an already skipped category seems harmless to me. And we could simply remove skipped categories from `onlyCategories`. Only if all of `onlyCategories` are skipped would we consider that a validation error, because that leaves us with nothing to run. But other than that, I would just log a warning. The "skipping already skipped" warning should probably be behind verbose flag as well. :thinking: